ethereum/contracts/pyth/PythUpgradable.sol

```diff
@@ -21,6 +23,44 @@ contract PythUpgradable is Initializable, OwnableUpgradeable, UUPSUpgradeable, P
         Pyth.initialize(wormhole, pyth2WormholeChainId, pyth2WormholeEmitter);
     }
 
+    // Migration call for upgrading to support the multiple price data sources feature
+    function migrateMultiSources() onlyOwner public {
+
+        require(!PythGetters.switchedToMultiSources(), "Already migrated to multiple data sources");
+        
+        addDataSource(PythInternalStructs.DataSource(PythGetters.pyth2WormholeChainId(), PythGetters.pyth2WormholeEmitter()));
+
+        require(PythGetters.isValidDataSource(PythGetters.pyth2WormholeChainId(), PythGetters.pyth2WormholeEmitter()), "Could not confirm migrated valid data source");
+        
+    }
```

Comment:
I wanted to make sure it's impossible to call this more than once

Reply:
I like Tom's idea because it makes contract code smaller.

Tom is suggesting to remove whole `migrateMultiSource` function (and its flag) from the contract and in implement the logic in the migration file that does the upgrade. It will be calling `addDataSource` after the update which is done by the owner as well and will be executed only when this migration step is running.

Reply:
Oh, I think that does make more sense. Thanks!